index-advisor.md

```diff
@@ -0,0 +1,223 @@
+
+For example, the `/*+ HYPO_INDEX(t, idx_ab, a, b) */` comment instructs the query planner to create a hypothetical index named `idx_ab` on table `t` for columns `a` and `b`. The planner generates the index's metadata but does not physically materialize it. If applicable, the planner considers this hypothetical index during query optimization without incurring the costs associated with index creation.
+
+The `RECOMMEND INDEX` advisor uses hypothetical indexes for "What-If" analysis to evaluate potential benefits of different indexes. You can also use hypothetical indexes directly to experiment with index designs before proceeding to create them.
+
+The following example shows a query using a hypothetical index:
+
+```sql
+CREATE TABLE t(a INT, b INT, c INT);
+Query OK, 0 rows affected (0.02 sec)
+
+EXPLAIN FORMAT='verbose' SELECT a, b FROM t WHERE a=1 AND b=1;
++-------------------------+----------+------------+-----------+---------------+----------------------------------+
+| id                      | estRows  | estCost    | task      | access object | operator info                    |
++-------------------------+----------+------------+-----------+---------------+----------------------------------+
+| TableReader_7           | 0.01     | 392133.42  | root      |               | data:Selection_6                 |
+| └─Selection_6           | 0.01     | 5882000.00 | cop[tikv] |               | eq(test.t.a, 1), eq(test.t.b, 1) |
+|   └─TableFullScan_5     | 10000.00 | 4884000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo   |
++-------------------------+----------+------------+-----------+---------------+----------------------------------+
+
+EXPLAIN FORMAT='verbose' SELECT /*+ HYPO_INDEX(t, idx_ab, a, b) */ a, b FROM t WHERE a=1 AND b=1;
++------------------------+---------+---------+-----------+-----------------------------+-------------------------------------------------+
+| id                     | estRows | estCost | task      | access object               | operator info                                   |
++------------------------+---------+---------+-----------+-----------------------------+-------------------------------------------------+
+| IndexReader_6          | 0.10    | 2.20    | root      |                             | index:IndexRangeScan_5                          |
+| └─IndexRangeScan_5     | 0.10    | 20.35   | cop[tikv] | table:t, index:idx_ab(a, b) | range:[1 1,1 1], keep order:false, stats:pseudo |
++------------------------+---------+---------+-----------+-----------------------------+-------------------------------------------------+
+```
+
+In this example, the `HYPO_INDEX` hint specifies a hypothetical index. Using this index reduces the estimated cost from `392133.42` to `2.20` by enabling an index range scan (`IndexRangeScan`) instead of a full table scan (`TableFullScan`).
```

Comment:
```suggestion
In this example, the `HYPO_INDEX` comment specifies a hypothetical index. Using this index reduces the estimated cost from `392133.42` to `2.20` by enabling an index range scan (`IndexRangeScan`) instead of a full table scan (`TableFullScan`).
```